lib/chef/knife/cloud/google_service.rb

```diff
@@ -186,12 +186,13 @@ def list_project_quotas
     def validate_server_create_options!(options)
       raise "Invalid machine type: #{options[:machine_type]}" unless valid_machine_type?(options[:machine_type])
       raise "Invalid network: #{options[:network]}" unless valid_network?(options[:network])
+      raise "Invalid subnet: #{options[:subnet]}" if options[:subnet] && !valid_subnet?(options[:subnet])
       raise "Invalid Public IP setting: #{options[:public_ip]}" unless valid_public_ip_setting?(options[:public_ip])
       raise "Invalid image: #{options[:image]} - check your image name, or set an image project if needed" if image_search_for(options[:image], options[:image_project]).nil?
     end
 
     def check_api_call(&block)
-      block.call
```

Comment:
This is interesting.  Is it more idiomatic to use `block.call` instead of `yield` when the block is explicitly defined as an instance variable `Proc`?  As I understand it there are two solid reasons for binding the block into a `Proc` instance: You need to check the arity of the block or you're going to call the block on another object.  Unless you need fall into either of those It's more performant to omit the variable binding and simply `yield if block_given?`.